kernel_gateway/services/kernels/handlers.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import notebook.services.kernels.handlers as notebook_handlers
+
+class TokenAuthorizationMixin(object):
+    '''
+    Adds token authentication to tornado.web.RequestHandlers.
+    '''
+    def prepare(self):
+        '''
+        If kg_auth_token is present in the application settings,
+        raises 401 if the request Authorization header does not contain
+        the value "Token <kg_auth_token>".
+        '''
+        server_token = self.settings.get('kg_auth_token')
```

Comment:
I prefixed the settings values to avoid conflicts with Tornado settings. I don't plan to propagate them further.

For the moment, I won't add the properties because I know for the CORS settings (next PR) I'm already going to have to refactor to handle the Websocket handler differently than the HTTP ones. I'll see what makes sense then.